fairlearn/metrics/_streaming_metric_frame.py

```diff
@@ -0,0 +1,182 @@
+            The predictions.
+
+        sensitive_features : (List, pandas.Series, dict of 1d arrays,
+         numpy.ndarray, pandas.DataFrame)
+            The sensitive features which should be used to create the subgroups.
+            At least one sensitive feature must be provided.
+            All names (whether on pandas objects or dictionary keys) must be strings.
+            We also forbid DataFrames with column names of ``None``.
+            For cases where no names are provided we generate names ``sensitive_feature_[n]``.
+
+        control_features : (List, pandas.Series, dict of 1d arrays,
+         numpy.ndarray, pandas.DataFrame)
+            Control features are similar to sensitive features, in that they
+            divide the input data into subgroups.
+            Unlike the sensitive features, aggregations are not performed
+            across the control features - for example, the ``overall`` property
+            will have one value for each subgroup in the control feature(s),
+            rather than a single value for the entire data set.
+            Control features can be specified similarly to the sensitive features.
+            However, their default names (if none can be identified in the
+            input values) are of the format ``control_feature_[n]``.
+
+        sample_params : dict
+            Parameters for the metric function(s). If there is only one metric function,
+            then this is a dictionary of strings and array-like objects, which are split
+            alongside the ``y_true`` and ``y_pred`` arrays, and passed to the metric function.
+            If there are multiple metric functions (passed as a dictionary), then this is
+            a nested dictionary, with the first set of string keys identifying the
+            metric function name, with the values being the string-to-array-like dictionaries.
+
+            **Note** the types returned by members of the class vary based on whether
+            control features are present.
+        """
+        check_consistent_length(y_true, y_pred)
+        self._y_true = self._concat_if_not_none(self._y_true, y_true)
+        self._y_pred = self._concat_if_not_none(self._y_pred, y_pred)
+        self._sensitive_features = self._concat_if_not_none(self._sensitive_features,
+                                                            sensitive_features)
+        if control_features is not None:
+            self._control_features = self._concat_if_not_none(self._control_features,
+                                                              control_features)
+        elif self._control_features is not None:
+            raise ValueError(_CF_BAD_STATE_ERR)
+
+        if sample_params is not None:
+            self._sample_params = self._concat_if_not_none(self._sample_params,
+                                                           sample_params)
+
+    def get_metric(self) -> MetricFrame:
```

Comment:
Perhaps `evaluate_metrics()` ? @MiroDudik @hildeweerts @adrinjalali ?

Reply:
that's a good question. Somehow I was thinking of this being sth that I could add data to, and then call `.evaluate()` which either returns nothing or `self`, and it would have all the features like `MetrifFrame`. If we don't want to go down that path, I think `get_metricframe` would be what this does?